SiReCore/utterance.py

```diff
@@ -432,3 +448,66 @@ def get_next_word(self):
     else:
       #We want the next one and have already added one to pos.
       return self.parent_utt.words[pos]
+
+  #Assigns emphasis to word if original text is capitalised
+  def get_emph(self):
+      if re.search(self.upper_reg, self.id) != None:
+          return True
+
+# Check if the next word is emphasised
+  def forward_emph(self):
+      pos = self.pos_in_utt() + 1
+      if pos == len(self.parent_utt.words):
+        return "xx"
+      else:
+          if re.search(self.upper_reg, self.parent_utt.words[pos].id) != None:
+              return True
+
+# Check if the previous word is emphasised
+  def backward_emph(self):
+      if self.pos_in_utt() == 0:
+        return "xx"
+      else:
+        pos = self.pos_in_utt() - 1
+        if re.search(self.upper_reg, self.parent_utt.words[pos].id) != None:
+            return True
+
+# Count how many words until the next emphasised word
+  def next_emph(self):
+      pos = self.pos_in_utt()
+      for i in xrange(pos, len(self.parent_utt.words)+1):
+          if i == len(self.parent_utt.words):
+            return "xx"
+          elif re.search(self.upper_reg, self.parent_utt.words[i].id) != None:
+            num = i - pos
+            # if the current word is the emphasised word, check if there is a next emphasised word.
```

Comment:
You should start your i at pos+1 in the first to not do the double for loop. There is no need for the first loop in fact.